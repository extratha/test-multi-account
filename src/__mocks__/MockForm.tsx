import { yupResolver } from "@hookform/resolvers/yup";
import { ReactNode, useEffect } from "react";
import { FormProvider, UseFormReturn, useForm } from "react-hook-form";
import { ObjectSchema } from "yup";

export interface MockFormProps<T = any> {
  initValues: T | undefined;
  resolver?: ObjectSchema<any>;
  children?: ReactNode;
  onInitForm?: (methods: UseFormReturn) => void;
  onSubmit: () => void;
}

export interface MockHandleValueProps {
  name: string;
}

const MockForm = (props: MockFormProps) => {
  const { initValues, resolver, children, onInitForm, onSubmit } = props;
  const methods = useForm({
    reValidateMode: "onSubmit",
    defaultValues: initValues,
    resolver: resolver ? yupResolver(resolver) : undefined,
  });

  useEffect(() => {
    if (onInitForm) onInitForm(methods);
  }, []);

  return (
    <FormProvider {...methods}>
      <form noValidate onSubmit={methods.handleSubmit(onSubmit)}>
        {children}
        <button type="submit" data-testid="submit-button">
          Submit
        </button>
      </form>
    </FormProvider>
  );
};

export default MockForm;
