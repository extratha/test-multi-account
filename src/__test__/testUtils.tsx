import { RenderOptions, RenderResult, render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { SWRConfig } from 'swr';
import { theme } from '@/config/config-mui/theme';
import { ThemeProvider as MUIThemeProvider } from '@mui/material/styles';
import { NextIntlClientProvider } from 'next-intl';
import commonTh from '../../public/locales/th/common.json';
import commonEn from '../../public/locales/th/common.json';
type RenderWithProvider = (
  elm: React.ReactElement,
  renderOptions?: ProviderOptions,
) => RenderResult;

export type ProviderOptions = {
  formOptions?: Record<string, unknown>;
} & RenderOptions;

const messages = {
  ...commonTh,
  ...commonEn,
};

const renderWithProviders: RenderWithProvider = (
  component,
  renderOptions?,
  locale = 'th',
) => {
  const wrapper = ({ children }: { children: React.ReactNode }) => (
    <NextIntlClientProvider locale={locale} messages={messages}>
      <MUIThemeProvider theme={theme}>
        <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      </MUIThemeProvider>
    </NextIntlClientProvider>
  );

  return render(component, {
    wrapper,
    ...renderOptions,
  });
};

export { renderWithProviders as render, userEvent };
export * from '@testing-library/react';
