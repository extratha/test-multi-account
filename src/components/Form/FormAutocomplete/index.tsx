import { CUSTOM_COLORS } from "@/config/config-mui/theme/colors";
import { KeyboardArrowDown } from "@mui/icons-material";
import { ListItem, Autocomplete as MuiAutocomplete, styled, TextField, Typography } from "@mui/material";
import { useTranslations } from "next-intl";
import { useController, useFormContext } from "react-hook-form";
import { FieldErrorMessage } from "../FieldErrorMessage";

export interface Option {
  label: string;
  value: string;
}

export interface FormAutocompleteProps {
  name: string;
  label: string;
  options?: Option[];
  required?: boolean;
  placeholder?: string;
}

const OptionItem = styled(ListItem)(({ theme }) => ({
  padding: "16px 24px",
  "&:hover": {
    cursor: "pointer",
    backgroundColor: theme.palette.grey[100],
  },
  "&&[aria-selected=true], &[aria-selected=true].Mui-focused, .Mui-focused": {
    backgroundColor: theme.palette.background.paper,
    "&:hover": {
      backgroundColor: theme.palette.grey[100],
    },
  },
}));
interface AutocompleteCustomStyleProps {
  onChange: (event: React.ChangeEvent<"object">, data: Option | null) => void;
}

export const Autocomplete = styled(MuiAutocomplete)<AutocompleteCustomStyleProps>(() => ({
  minWidth: "200px",
}));

const FormAutocomplete = ({ name, options, required = false, placeholder }: FormAutocompleteProps) => {
  const t = useTranslations("Common");

  const { control } = useFormContext();
  const { field, fieldState } = useController({ name, control });
  const error = fieldState.error?.message || "";

  return (
    <>
      <Autocomplete
        {...field}
        value={field.value || ""}
        data-testid={`autocomplete-${name}`}
        options={options || []}
        // TODO : Refactor
        isOptionEqualToValue={(option: any, value: any) => option.id === value.id}
        noOptionsText={t("text.noOptions")}
        renderInput={(params) => (
          <TextField
            {...params}
            required={required}
            error={!!fieldState.error}
            placeholder={`${placeholder || t("placeholder.selectValue")}`}
          />
        )}
        renderOption={(props, option) => (
          <OptionItem {...props} key={(option as Option).value}>
            <Typography variant="bodyLarge" color={CUSTOM_COLORS.lightSteelgray}>
              {(option as Option).label}
            </Typography>
          </OptionItem>
        )}
        onChange={(_, data) => {
          console.log("Selected data:", data);
          field.onChange((data as Option)?.value || null);
        }}
        popupIcon={<KeyboardArrowDown data-testid="drop-icon" />}
      />
      {error && (
        <FieldErrorMessage data-testid={`error-field-${name}`} variant="bodyLarge">
          {error}
        </FieldErrorMessage>
      )}
    </>
  );
};

export default FormAutocomplete;
