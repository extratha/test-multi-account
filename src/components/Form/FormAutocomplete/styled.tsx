import { Autocomplete } from "@mui/material";
import { styled } from "@mui/material/styles";
import { Option } from ".";

interface AutocompleteCustomStyleProps {
  getOptionLabel: (option: Option) => string;
  onChange: (event: React.ChangeEvent<"object">, data: Option | null) => void;
}

export const AutocompleteCustomStyle = styled(Autocomplete)<AutocompleteCustomStyleProps>(() => ({
  minWidth: "200px",
}));
