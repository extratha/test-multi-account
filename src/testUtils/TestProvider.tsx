import { ThemeProvider } from "@emotion/react";

import { ReactNode } from "react";
import { SWRConfig } from "swr";

import ModalLayer from "@/components/Modal/ModalLayer";
import { theme } from "@/config/config-mui";

interface TestProviderProps {
  children?: ReactNode;
}

export const TestProvider = ({ children }: TestProviderProps) => {
  return (
    <ThemeProvider theme={theme}>
      <SWRConfig value={{ provider: () => new Map() }}>{children}</SWRConfig>
      <ModalLayer />
    </ThemeProvider>
  );
};
