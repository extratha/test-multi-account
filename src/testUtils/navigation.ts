import * as NextNavigation from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
  useParams: jest.fn(),
  usePathname: jest.fn(),
  useSearchParams: jest.fn(),
  useServerInsertedHTML: jest.fn(),
}));

export interface SpyUseSearchParams {
  get: jest.Mock;
}

export interface SpyUseRouter {
  replace: jest.Mock;
  push: jest.Mock;
  back: jest.Mock;
}

export const spyUseParams = (params: any = {}) => {
  jest.spyOn(NextNavigation, "useParams").mockReturnValue(params);
  return NextNavigation.useParams as jest.Mock;
};

export const spyUseRouter = (): SpyUseRouter => {
  const replace = jest.fn();
  const push = jest.fn();
  const back = jest.fn();
  jest.spyOn(NextNavigation, "useRouter").mockReturnValue({ replace, push, back } as any);
  return { replace, push, back };
};

export const spyUsePathname = (pathname: string) => {
  jest.spyOn(NextNavigation, "usePathname").mockReturnValue(pathname);
};

export const spyUseSearchParams = (): SpyUseSearchParams => {
  const get = jest.fn();
  jest.spyOn(NextNavigation, "useSearchParams").mockReturnValue({ get } as any);
  return {
    get,
  };
};

export const spyUseServerInsertedHTML = () => {
  return jest.spyOn(NextNavigation, "useServerInsertedHTML");
};
