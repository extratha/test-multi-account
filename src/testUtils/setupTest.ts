import "@testing-library/jest-dom";
import "jest-canvas-mock";

import "@/__mocks__/Markdown";
import "@/__mocks__/MockFirebase";
import "@/__mocks__/NextImage";
import { LOCALE_LANGUAGE } from "@/config";
import { initI18next } from "@/locales/i18n";
import locale from "@/locales/th.json";

initI18next(LOCALE_LANGUAGE.TH, locale);

jest.mock("@/api/api", () => {
  const actualModule = jest.requireActual("@/api/api");
  return { ...actualModule, __esModule: true };
});

jest.mock("@/api/apiUnauthorize", () => {
  const actualModule = jest.requireActual("@/api/apiUnauthorize");
  return { ...actualModule, __esModule: true };
});

jest.mock("next/config", () => () => ({
  setConfig: jest.fn(),
  publicRuntimeConfig: {
    apiToken:
      "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3MDY2NzUzNDQsImlhdCI6MTcwNjU4ODk0NCwicHJvZmlsZUlkIjoiZDU1YWY3NDUtOTYyZS00MTk2LTkyOTEtZDg2YTliYWNiZWY2In0.sNgwYjIGLYEVAptqILGgRVGXwt9FCPKHRubEZyaodhg",
    mockBaseApiUrl: "https://www.mock-playground-dashboard.co.th",
    baseApiUrl: "https://www.mock-playground-dashboard.co.th",
    globalEndpoint: "playground-dashboard",
    publicApiHost: "http://localhost:3000/",
  },
}));

jest.mock("cookies-next", () => ({
  __esModule: true,
  ...jest.requireActual("cookies-next"),
  getCookie: jest.fn().mockReturnValue("MOCK_ACCESS_TOKEN"),
}));
