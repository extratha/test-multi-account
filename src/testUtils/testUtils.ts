import { act, render as testRender } from "@testing-library/react";
import dayjs from "dayjs";
import flushPromises from "flush-promises";
import { ReactElement } from "react";

import { TestProvider } from "./TestProvider";

export const API = {
  LOGIN: "/auth/login",
  EXAMPLE_AI_INTERPRET_URL: "/lab/examples/labId",
  GET_LAB_INTERPRETS_ID_URL: "/lab/interprets/labId",
  SUBMIT_LAB_INTERPRETS_URL: "/lab/interprets",
  CONSENT: "/consents",
  TERMS_AND_CONDITIONS: "/consents/terms-conditions",
  TERMS_AND_CONDITIONS_LATEST: "/consents/terms-conditions/latest",
  PRIVACY_POLICY: "/consents/privacy-policies",
  PRIVACY_POLICY_LATEST: "/consents/privacy-policies/latest",
  PUBLIC_TERMS_CONDITIONS: "/consents/terms-conditions/public",
  PUBLIC_PRIVACY_POLICIES: "/consents/privacy-policies/public",
};

export const render = (ui: ReactElement) => {
  return testRender(ui, { wrapper: TestProvider });
};

export const flushPromise = async () => {
  await act(async () => {
    await flushPromises();
  });
};

export const advanceTimersByTime = async (time: number) => {
  await act(async () => {
    jest.advanceTimersByTime(time);
  });
};

export const setMockDate = (date: string) => {
  jest.setSystemTime(dayjs(date).valueOf());
};
