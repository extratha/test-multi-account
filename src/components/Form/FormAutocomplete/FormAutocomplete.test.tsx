import MockForm, { MockFormProps } from "@/__tests__/__mocks__/MockForm";
import * as yup from "yup";
import FormAutocomplete, { FormAutocompleteProps } from ".";
import { fireEvent, flushPromise, render, screen, userEvent } from "../../../__tests__/testUtils";
describe("Autocomplete", () => {
  const props: FormAutocompleteProps = {
    name: "test",
    label: "Test",
    options: [
      {
        label: "test1",
        value: "Test1",
      },
      {
        label: "test2",
        value: "Test2",
      },
    ],
  };
  let form: MockFormProps;
  const error = {
    message: {
      require: "this field is require",
      invalid: "invalid value",
    },
  };
  let fieldSchema = yup.string().nullable();
  const renderComponent = async (props: FormAutocompleteProps) => {
    if (props.required) {
      fieldSchema = fieldSchema.required(error.message.require);
    }
    if (props.options) {
      fieldSchema = fieldSchema.oneOf(
        props.options.map((value) => value.value),
        error.message.invalid
      );
    }
    form = {
      initValues: { test: "" },
      resolver: yup.object({ test: fieldSchema }),
      onSubmit: jest.fn(),
    };
    const view = render(
      <MockForm {...form}>
        <FormAutocomplete {...props} />
      </MockForm>
    );
    await flushPromise();
    return view;
  };
  it("should render correctly", async () => {
    const { asFragment } = await renderComponent(props);
    expect(asFragment()).toMatchSnapshot();
  });
  it("should display correctly", async () => {
    await renderComponent(props);
    expect(screen.getByTestId("autocomplete-test")).toBeInTheDocument();
  });
  it("should set value when select.", async () => {
    await renderComponent(props);
    const autocomplete = screen.getByRole("combobox");
    fireEvent.click(autocomplete);
    fireEvent.change(autocomplete, { target: { value: "test1" } });
    expect(autocomplete).toHaveValue("test1");
  });

  it("should has empty text when try to input value not in options.", async () => {
    await renderComponent(props);

    const autocomplete = screen.getByRole("combobox");
    fireEvent.focus(autocomplete);
    fireEvent.change(autocomplete, { target: { value: "xxxx" } });
    const dropdown = screen.getByText(/ไม่พบข้อมูล/i);
    expect(dropdown).toBeInTheDocument();
  });

  it("should display options correctly", async () => {
    await renderComponent(props);

    const autoComplete = screen.getByRole("combobox");
    await userEvent.click(autoComplete);

    expect(screen.getByText("test1")).toBeInTheDocument();
    expect(screen.getByText("test2")).toBeInTheDocument();
  });
  it("should clear value", async () => {
    await renderComponent(props);
    const autocomplete = screen.getByRole("combobox");
    fireEvent.click(autocomplete);
    fireEvent.change(autocomplete, { target: { value: "test1" } });
    const clear = screen.getByTitle("Clear");
    await userEvent.click(clear);
    expect(autocomplete).toHaveValue("");
  });

  it("should render with no options.", async () => {
    props.options = undefined;
    await renderComponent(props);

    const autocomplete = screen.getByRole("combobox");

    expect(autocomplete).toBeInTheDocument();
  });
});
