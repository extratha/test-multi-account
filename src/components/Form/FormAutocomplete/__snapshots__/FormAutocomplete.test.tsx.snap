// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAutocomplete should render correctly 1`] = `
<DocumentFragment>
  .emotion-0.Mui-focused .MuiAutocomplete-clearIndicator {
  visibility: visible;
}

@media (pointer: fine) {
  .emotion-0:hover .MuiAutocomplete-clearIndicator {
    visibility: visible;
  }
}

.emotion-0 .MuiAutocomplete-tag {
  margin: 3px;
  max-width: calc(100% - 6px);
}

.MuiAutocomplete-hasPopupIcon.emotion-0 .MuiAutocomplete-inputRoot,
.MuiAutocomplete-hasClearIcon.emotion-0 .MuiAutocomplete-inputRoot {
  padding-right: 30px;
}

.MuiAutocomplete-hasPopupIcon.MuiAutocomplete-hasClearIcon.emotion-0 .MuiAutocomplete-inputRoot {
  padding-right: 56px;
}

.emotion-0 .MuiAutocomplete-inputRoot .MuiAutocomplete-input {
  width: 0;
  min-width: 30px;
}

.emotion-0 .MuiInput-root {
  padding-bottom: 1px;
}

.emotion-0 .MuiInput-root .MuiInput-input {
  padding: 4px 4px 4px 0px;
}

.emotion-0 .MuiInput-root.MuiInputBase-sizeSmall .MuiInput-input {
  padding: 2px 4px 3px 0;
}

.emotion-0 .MuiOutlinedInput-root {
  padding: 9px;
}

.MuiAutocomplete-hasPopupIcon.emotion-0 .MuiOutlinedInput-root,
.MuiAutocomplete-hasClearIcon.emotion-0 .MuiOutlinedInput-root {
  padding-right: 39px;
}

.MuiAutocomplete-hasPopupIcon.MuiAutocomplete-hasClearIcon.emotion-0 .MuiOutlinedInput-root {
  padding-right: 65px;
}

.emotion-0 .MuiOutlinedInput-root .MuiAutocomplete-input {
  padding: 7.5px 4px 7.5px 5px;
}

.emotion-0 .MuiOutlinedInput-root .MuiAutocomplete-endAdornment {
  right: 9px;
}

.emotion-0 .MuiOutlinedInput-root.MuiInputBase-sizeSmall {
  padding-top: 6px;
  padding-bottom: 6px;
  padding-left: 6px;
}

.emotion-0 .MuiOutlinedInput-root.MuiInputBase-sizeSmall .MuiAutocomplete-input {
  padding: 2.5px 4px 2.5px 8px;
}

.emotion-0 .MuiFilledInput-root {
  padding-top: 19px;
  padding-left: 8px;
}

.MuiAutocomplete-hasPopupIcon.emotion-0 .MuiFilledInput-root,
.MuiAutocomplete-hasClearIcon.emotion-0 .MuiFilledInput-root {
  padding-right: 39px;
}

.MuiAutocomplete-hasPopupIcon.MuiAutocomplete-hasClearIcon.emotion-0 .MuiFilledInput-root {
  padding-right: 65px;
}

.emotion-0 .MuiFilledInput-root .MuiFilledInput-input {
  padding: 7px 4px;
}

.emotion-0 .MuiFilledInput-root .MuiAutocomplete-endAdornment {
  right: 9px;
}

.emotion-0 .MuiFilledInput-root.MuiInputBase-sizeSmall {
  padding-bottom: 1px;
}

.emotion-0 .MuiFilledInput-root.MuiInputBase-sizeSmall .MuiFilledInput-input {
  padding: 2.5px 4px;
}

.emotion-0 .MuiInputBase-hiddenLabel {
  padding-top: 8px;
}

.emotion-0 .MuiFilledInput-root.MuiInputBase-hiddenLabel {
  padding-top: 0;
  padding-bottom: 0;
}

.emotion-0 .MuiFilledInput-root.MuiInputBase-hiddenLabel .MuiAutocomplete-input {
  padding-top: 16px;
  padding-bottom: 17px;
}

.emotion-0 .MuiFilledInput-root.MuiInputBase-hiddenLabel.MuiInputBase-sizeSmall .MuiAutocomplete-input {
  padding-top: 8px;
  padding-bottom: 9px;
}

.emotion-0 .MuiAutocomplete-input {
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  text-overflow: ellipsis;
  opacity: 0;
}

.emotion-0 .MuiAutocomplete-input {
  opacity: 1;
}

.emotion-0 .MuiAutocomplete-inputRoot {
  padding: 4px 8px;
}

.emotion-1 {
  display: inline-flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  position: relative;
  min-width: 0;
  padding: 0;
  margin: 0;
  border: 0;
  vertical-align: top;
  width: 100%;
}

.emotion-2 {
  font-family: fontFamily;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.4375em;
  color: rgba(0, 0, 0, 0.87);
  box-sizing: border-box;
  position: relative;
  cursor: text;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  border-radius: 8px;
  padding-right: 14px;
}

.emotion-2.Mui-disabled {
  color: #B0BEC5;
  cursor: default;
}

.emotion-2:hover .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.87);
}

@media (hover: none) {
  .emotion-2:hover .MuiOutlinedInput-notchedOutline {
    border-color: rgba(0, 0, 0, 0.23);
  }
}

.emotion-2.Mui-focused .MuiOutlinedInput-notchedOutline {
  border-color: #2196F3;
  border-width: 2px;
}

.emotion-2.Mui-error .MuiOutlinedInput-notchedOutline {
  border-color: #d32f2f;
}

.emotion-2.Mui-disabled .MuiOutlinedInput-notchedOutline {
  border-color: rgba(0, 0, 0, 0.26);
}

.emotion-3 {
  font: inherit;
  letter-spacing: inherit;
  color: currentColor;
  padding: 12px;
  border: 0;
  box-sizing: content-box;
  background: none;
  height: 1.4375em;
  margin: 0;
  -webkit-tap-highlight-color: transparent;
  display: block;
  min-width: 0;
  width: 100%;
  -webkit-animation-name: mui-auto-fill-cancel;
  animation-name: mui-auto-fill-cancel;
  -webkit-animation-duration: 10ms;
  animation-duration: 10ms;
  padding-right: 0;
}

.emotion-3::-webkit-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-moz-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3::-ms-input-placeholder {
  color: currentColor;
  opacity: 0.42;
  -webkit-transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: opacity 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
}

.emotion-3:focus {
  outline: 0;
}

.emotion-3:invalid {
  box-shadow: none;
}

.emotion-3::-webkit-search-decoration {
  -webkit-appearance: none;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-webkit-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-moz-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3::-ms-input-placeholder {
  opacity: 0!important;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-webkit-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-moz-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus:-ms-input-placeholder {
  opacity: 0.42;
}

label[data-shrink=false]+.MuiInputBase-formControl .emotion-3:focus::-ms-input-placeholder {
  opacity: 0.42;
}

.emotion-3.Mui-disabled {
  opacity: 1;
  -webkit-text-fill-color: #B0BEC5;
}

.emotion-3:-webkit-autofill {
  -webkit-animation-duration: 5000s;
  animation-duration: 5000s;
  -webkit-animation-name: mui-auto-fill;
  animation-name: mui-auto-fill;
}

.emotion-3:-webkit-autofill {
  border-radius: inherit;
}

.emotion-4 {
  position: absolute;
  right: 0;
  top: 50%;
  -webkit-transform: translate(0, -50%);
  -moz-transform: translate(0, -50%);
  -ms-transform: translate(0, -50%);
  transform: translate(0, -50%);
}

.emotion-5 {
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 50%;
  padding: 2px;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: rgba(0, 0, 0, 0.54);
  text-align: center;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  font-size: 1.5rem;
  overflow: visible;
  -webkit-transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  margin-right: -2px;
}

.emotion-5::-moz-focus-inner {
  border-style: none;
}

.emotion-5.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-5 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-5:hover {
  background-color: rgba(0, 0, 0, 0.04);
}

@media (hover: none) {
  .emotion-5:hover {
    background-color: transparent;
  }
}

.emotion-5.Mui-disabled {
  background-color: transparent;
  color: rgba(0, 0, 0, 0.26);
}

.emotion-6 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
}

.emotion-7 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-8 {
  text-align: left;
  position: absolute;
  bottom: 0;
  right: 0;
  top: -5px;
  left: 0;
  margin: 0;
  padding: 0 8px;
  pointer-events: none;
  border-radius: inherit;
  border-style: solid;
  border-width: 1px;
  overflow: hidden;
  min-width: 0%;
  border-color: rgba(0, 0, 0, 0.23);
}

.emotion-9 {
  float: unset;
  width: auto;
  overflow: hidden;
  padding: 0;
  line-height: 11px;
  -webkit-transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
  transition: width 150ms cubic-bezier(0.0, 0, 0.2, 1) 0ms;
}

<form
    novalidate=""
  >
    <div
      class="MuiAutocomplete-root MuiAutocomplete-hasPopupIcon emotion-0"
      data-testid="autocomplete-test"
      name="test"
    >
      <div
        class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root emotion-1"
      >
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot emotion-2"
        >
          <input
            aria-autocomplete="list"
            aria-expanded="false"
            aria-invalid="false"
            autocapitalize="none"
            autocomplete="off"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused emotion-3"
            id=":r0:"
            role="combobox"
            spellcheck="false"
            type="text"
            value=""
          />
          <div
            class="MuiAutocomplete-endAdornment emotion-4"
          >
            <button
              aria-label="Open"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiAutocomplete-popupIndicator emotion-5"
              tabindex="-1"
              title="Open"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium emotion-6"
                data-testid="drop-icon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M7.41 8.59 12 13.17l4.59-4.58L18 10l-6 6-6-6z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root emotion-7"
              />
            </button>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline emotion-8"
          >
            <legend
              class="emotion-9"
            >
              <span
                class="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <button
      data-testid="submit-button"
      type="submit"
    >
      Submit
    </button>
  </form>
</DocumentFragment>
`;
